pkg/apis/core/types.go

```diff
@@ -2378,6 +2378,12 @@ const (
 	TolerationOpEqual  TolerationOperator = "Equal"
 )
 
+// ReadinessGate contains the reference to a pod condition
+type ReadinessGate struct {
+	// ConditionType refers to a condition in the pod's condition list with matching type.
+	ConditionType PodConditionType
```

Comment:
Or should this be a string?
```go
ConditionType string
```

Reply:
ReadinessGate refers to a PodCondition. And PodConditionType is the key of the condition list. Although PodConditionType is a string